### PR TITLE
[RIP-9] Add API_Reference_DefaultMQProducer.md for Java

### PR DESCRIPTION
**Pull request by Childe-Chen**
## What is the purpose of the change

Add API_Reference_DefaultMQProducer.md for Java

## Brief changelog

Add API_Reference_DefaultMQProducer.md for Java

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 1]
```diff
@@ -0,0 +1,1087 @@
+## DefaultMQProducer
+---
+### 类简介
+
+`public class DefaultMQProducer 
+extends ClientConfig 
+implements MQProducer`
+
+>`DefaultMQProducer`类是应用用来投递消息的入口，开箱即用，可通过无参构造方法快速创建一个生产者。主要负责消息的发送，支持同步/异步/onway的发送方式，发送方式均支持批量发送。可以通过它提供有一系列的getter/setter方法，调整发送者的参数。`DefaultMQProducer`提供了多个send方法，每个send方法略有不同，在使用前务必详细了解其意图。下面给出一个生产者示例代码，[点击查看更多示例代码](https://github.com/apache/rocketmq/blob/master/example/src/main/java/org/apache/rocketmq/example/)。
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: oneway

Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: oneway

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 2]
```diff
@@ -0,0 +1,1087 @@
...
+|SendResult|send(Message msg)|同步单条发送消息|
+|SendResult|send(Message msg, long timeout)|同步发送单条消息，并指定超时时间|
+|SendResult|send(Message msg, MessageQueue mq)|向指定的消息队列同步发送单条消息|
+|SendResult|send(Message msg, MessageQueue mq, long timeout)|向指定的消息队列同步单条发送消息，并指定超时时间|
+|void|send(Message msg, MessageQueue mq, SendCallback sendCallback)|向指定的消息队列异步单条发送消息，并指定回调方法|
+|void|send(Message msg, MessageQueue mq, SendCallback sendCallback, long timeout)|向指定的消息队列异步单条发送消息，并指定回调方法和超时时间|
+|SendResult|send(Message msg, MessageQueueSelector selector, Object arg)|向消息队列同步单条发送消息，并指定发送队列选择器|
+|SendResult|send(Message msg, MessageQueueSelector selector, Object arg, long timeout)|向消息队列同步单条发送消息，并指定发送队列选择器与超时时间|
+|void|send(Message msg, MessageQueueSelector selector, Object arg, SendCallback sendCallback)|向指定的消息队列异步单条发送消息|
+|void|send(Message msg, MessageQueueSelector selector, Object arg, SendCallback sendCallback, long timeout)|向指定的消息队列异步单条发送消息|
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: 向指定的消息队列异步单条发送消息，并指定超时时间

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 3]
```diff
@@ -0,0 +1,1087 @@
...
+|void|send(Message msg, SendCallback sendCallback)|异步发送消息|
+|void|send(Message msg, SendCallback sendCallback, long timeout)|异步发送消息，并指定回调方法和超时时间|
+|TransactionSendResult|sendMessageInTransaction(Message msg, LocalTransactionExecuter tranExecuter, final Object arg)|发送事务消息|
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: 发送事务消息，并指定本地执行事务实例

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 4]
```diff
@@ -0,0 +1,1087 @@
...
+
+- retryTimesWhenSendFailed
+
+	`private int retryTimesWhenSendFailed = 2`
+
+	同步模式下，在返回发送失败之前，内部尝试重新发送消息的最大次数。
+
+	默认值：2，即：默认情况下一条消息最多会被投递3次。
+
+	注意：这可能会导致消息的重复。
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: 在极端情况下，这可能会导致消息的重复。

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 5]
```diff
@@ -0,0 +1,1087 @@
...
+	`DefaultMQProducer(final String producerGroup, boolean enableMsgTrace, final String customizedTraceTopic)`
+
+	使用指定的分组名创建一个生产者，并设置是否开启消息追踪及追踪topic的名称。
+
+	- 入参描述：
+
+		参数名 | 类型 | 是否必须 | 缺省值 |描述
+		---|---|---|---|---
+		producerGroup | String | 是 | DEFAULT_PRODUCER | 生产者的分组名称
+		rpcHook | RPCHook | 否 | null |每个远程命令执行后会回掉rpcHook
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: 回调

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 6]
```diff
@@ -0,0 +1,1087 @@
...
+		InterruptedException - 发送线程中断。<br>
+		RemotingTooMuchRequestException - 发送超时。
+
+10. send
+
+	`public SendResult send(Collection<Message> msgs, long timeout)`
+
+	同步批量发送消息，如果在指定的超时时间内未完成消息投递，会抛出*RemotingTooMuchRequestException*。在返回发送失败之前，内部尝试重新发送消息的最大次数（参见*retryTimesWhenSendFailed*属性），这可能会导致消息的重复。未明确指定发送队列，默认采取轮询策略发送。
+
+	注意：指定队列意味着所有消息均为同一个topic。
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: 接口无法指定队列

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 7]
```diff
@@ -0,0 +1,1087 @@
...
+
+		MQClientException - 生产者状态非Running；没有找到broker；broker返回失败；网络异常；线程中断等客户端异常。
+
+10. send
+
+	`public SendResult send(Collection<Message> msgs)`
+
+	同步批量发送消息，如果在既定的超时时间内未完成消息投递，会抛出*RemotingTooMuchRequestException*。在返回发送失败之前，内部尝试重新发送消息的最大次数（参见*retryTimesWhenSendFailed*属性），这可能会导致消息的重复。未明确指定发送队列，默认采取轮询策略发送。
+	
+	注意：指定队列意味着所有消息均为同一个topic。
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: 接口无法指定队列

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 8]
```diff
@@ -0,0 +1,1087 @@
...
+		RemotingException - 网络异常。<br>
+		MQBrokerException - broker发生错误。<br>
+		InterruptedException - 发送线程中断。<br>
+		RemotingTooMuchRequestException - 发送超时。
+
+15. send
+
+	`send(Message msg, MessageQueue mq)`
+
+	向指定的消息队列同步发送单条消息，如果在指定的超时时间内未完成消息投递，会抛出*RemotingTooMuchRequestException*。仅当发送过程完全完成时，此方法才会返回。
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: 这个接口并没有指定超时时间

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md [hunk 9]
```diff
@@ -0,0 +1,1087 @@
...
+		RemotingException - 网络异常。<br>
+		MQBrokerException - broker发生错误。<br>
+		InterruptedException - 发送线程中断。<br>
+		RemotingTooMuchRequestException - 发送超时。
+
+17. send
+
+	`public void send(Message msg, MessageQueue mq, SendCallback sendCallback)`
+
+	向指定的消息队列异步发送单条消息，异步发送调用后直接返回，并在在发送成功或者异常时回调`sendCallback`，所以异步发送时`sendCallback`参数不能为null，否则在回调时会抛出`NullPointerException`。异步发送时，在成功发送前，其内部会尝试重新发送消息的最大次数（参见*retryTimesWhenSendAsyncFailed*属性），这有可能会导致消息的重复。
```
Comment (wlliqipeng) [inline docs/cn/client/java/API_Reference_DefaultMQProducer.md]: “这有可能会导致消息的重复”这句话都去掉

File: docs/cn/client/java/API_Reference_DefaultMQProducer.md (hunk 5 above)
Comment (Childe-Chen) [reply docs/cn/client/java/API_Reference_DefaultMQProducer.md]: 错别字会全局替换

**Pull request merged**